### PR TITLE
feat: Add basic session saving

### PR DESCRIPTION
**Pull request by cjh1**
This currently saves/restores the datasets and the currently selected dataset.

### CODE REVIEW

File: src/io/io.ts [hunk 1]
```diff
@@ -89,27 +90,35 @@ export async function retypeFile(file: File): Promise<File> {
...
   const results = await Promise.all(
     files.map(async (file) => {
       if (file.type === 'zip') {
         return extractFilesFromZip(file);
       }
-      return [file];
+
+      const entry: FileEntry = { file, path: '' };
+
+      return [entry];
```
Comment (floryst) [inline src/io/io.ts:102]: Could just collapse this into `[{ file, path: '' }]`.

File: src/io/io.ts [hunk 2]
```diff
@@ -89,27 +90,35 @@ export async function retypeFile(file: File): Promise<File> {
 
 export async function extractArchivesRecursively(
   files: File[]
-): Promise<File[]> {
+): Promise<FileEntry[]> {
   const results = await Promise.all(
     files.map(async (file) => {
       if (file.type === 'zip') {
```
Comment (floryst) [inline src/io/io.ts:96]: I think this also needs to be `ARCHIVE_FILE_TYPES.has(file.type)`, just to be sure.

File: src/components/SaveSession.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      </v-btn>
+    </v-card-actions>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from '@vue/composition-api';
+import { save } from '../io/state-file';
+
+const DEFAULT_FILENAME = 'volview.zip';
```
Comment (floryst) [inline src/components/SaveSession.vue:30]: Maybe `session.volview.zip`. We don't need to hide it being a zip file, but keeping a sub-extension of `volview` is helpful.

File: src/io/zip.ts [hunk 4]
```diff
@@ -1,19 +1,34 @@
...
+    const entries: FileEntry[] = [];
+    files.forEach((file, index) => {
+      entries.push({
+        file,
+        path: paths[index],
+      });
+    });
+
+    return entries;
+  });
```
Comment (floryst) [inline src/io/zip.ts:33]: This can just be a `files.map(....)`.

**Review by floryst — COMMENTED** [review]
Looks good so far! Just some minor things from my first pass.

File: src/components/SaveSession.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    <v-card-text>
+      <v-text-field
+        v-on:keyup.enter="saveSession"
+        v-model="fileName"
+        hint="The filename to use for the session state file."
+        label="Session State Filename"
+      />
+    </v-card-text>
+    <v-card-actions>
+      <v-spacer></v-spacer>
```
[the remark this answers is not in the document]
Comment (cjh1) [reply src/components/SaveSession.vue]: Indeed! Done

File: src/components/SaveSession.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import { save } from '../io/state-file';
+
+const DEFAULT_FILENAME = 'volview.zip';
+
+export default defineComponent({
+  setup(props, { emit }) {
+    const fileName = ref(DEFAULT_FILENAME);
+
+    async function saveSession() {
+      await save(fileName.value);
```
[the remark this answers is not in the document]
Comment (cjh1) [reply src/components/SaveSession.vue:41]: Done

File: src/io/io.ts (hunk 1 above)
Comment (cjh1) [reply src/io/io.ts]: Done

File: src/io/io.ts (hunk 2 above)
Comment (cjh1) [reply src/io/io.ts]: Done

File: src/components/SaveSession.vue (hunk 3 above)
Comment (cjh1) [reply src/components/SaveSession.vue]: Do we want to enforce the sub-extension?

File: src/io/zip.ts (hunk 4 above)
Comment (cjh1) [reply src/io/zip.ts]: Done

**Comment by floryst** [comment]
LGTM!

**Pull request merged**